designs/capi-resources-custom-namespace.md

```diff
@@ -0,0 +1,54 @@
+# CAPI Resources in Custom Namespace
+
+## Introduction
+
+**Problem:** Customers managing large scale EKS-A clusters (hundreds or thousands of workload clusters for a given management cluster) expect to have difficulty managing all these resources if they are all in the same namespace 
+
+This document aims to solve that, by allowing customers to specify a new flag on the cluster spec. If set, the underlying CAPI/CAPX/etcd resources will be created under the Cluster's namespace, rather than all dumped into `eksa-system`. 
+This change would make it so customers would be able to put these resources virtually in any namespace, despite the fact that they may be considered eks-a internal components.
+
+### Tenets
+
+****Simple:**** make everything simple and secure by default, but give options to change those defaults if there is a need
+
+### Goals and Objectives
+
+As a EKS-A cluster administrator I want to organize/partition each cluster into a dedicated namespace so that I can more easily manage my workload clusters
+
+### Statement of Scope
+
+**In scope**
+* Allow users to indicate whether they'd like all underlying cluster resources to fall into the Cluster's namespace
+
+**Not in scope**
+* Support for customizing which resources are going into which namespace at different layers (e.g. CAPC resources into ns1, CAPI resources into ns2, eks-a resources into ns3, etc.)
+
+**Future scope**
+* N/A
+ 
+## Current state
+1. Top level eks-a resources (e.g. Cluster, *DatacenterConfig, *MachineConfig) all go into the "default" namespace by default but can be customized to any namespace. Ideally, customers should only ever interact with these top-level eks-a resources
+2. Underlying CAPI/CAPX and etcd resources are created in the eksa-system, to be hidden from customers as implementation details
+
+## Overview of Solution
+1. Introduce a new boolean flag in the EKS-A Cluster CRD spec which allows users to indicate whether they would like the underlying resources to be created in the same namespace as the eks-a Cluster resource
```

Comment:
Instead of a boolean flag, what about introducing a new field in the cluster spec, like `systemNamespace` or something? Is the purpose of this that we just want some override functionality? Our controller would also need to know where to pull the resources, hence having it in the cluster spec would be easier.

Reply:
as a cluster administrator I'd like to be able to specify exactly which namespace the components go in (and have you stop me if I try to put it in default or kube-system)

Reply:
Will add this user story, thanks folks.